databricks/sdk/credentials_provider.py

```diff
@@ -664,6 +665,84 @@ def inner() -> Dict[str, str]:
     return inner
 
 
+# This Code is derived from Mlflow DatabricksModelServingConfigProvider
+# https://github.com/mlflow/mlflow/blob/1219e3ef1aac7d337a618a352cd859b336cf5c81/mlflow/legacy_databricks_cli/configure/provider.py#L332
+class ModelServingAuthProvider():
+    _MODEL_DEPENDENCY_OAUTH_TOKEN_FILE_PATH = "/var/credentials-secret/model-dependencies-oauth-token"
+
+    def __init__(self):
+        self.expiry_time = -1
+        self.current_token = None
+        self.refresh_duration = 300 # 300 Seconds
+
+    def should_fetch_model_serving_environment_oauth(self) -> bool:
+        """
+        Check whether this is the model serving environment
+        Additionally check if the oauth token file path exists
+        """
+
+        is_in_model_serving_env = (os.environ.get("IS_IN_DB_MODEL_SERVING_ENV")
+                                   or os.environ.get("IS_IN_DATABRICKS_MODEL_SERVING_ENV") or "false")
+        return (is_in_model_serving_env == "true"
+                and os.path.isfile(self.__class__._MODEL_DEPENDENCY_OAUTH_TOKEN_FILE_PATH))
+
+    def get_model_dependency_oauth_token(self, should_retry=True) -> str:
+        if self.current_token is not None and self.expiry_time > time.time():
+            return self.current_token
+        else:
+            try:
+                with open(self.__class__._MODEL_DEPENDENCY_OAUTH_TOKEN_FILE_PATH) as f:
+                    oauth_dict = json.load(f)
+                    self.current_token = oauth_dict["OAUTH_TOKEN"][0]["oauthTokenValue"]
+                    self.expiry_time = time.time() + self.refresh_duration
+            except Exception as e:
+                # sleep and retry in case of any race conditions with OAuth refreshing
+                if should_retry:
+                    time.sleep(0.5)
```

Comment:
Can you log the exception here? This will be useful debugging information if something goes wrong.